src/Stratis.Bitcoin/P2P/SelfEndpointTracker.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.P2P
+{
+    /// <summary>
+    /// Tracker for endpoints known to be self.
+    /// </summary>
+    public class SelfEndpointTracker : ISelfEndpointTracker
+    {
+        /// <summary>Stored endpoints expire after this many hours and need to be readded.
+        /// <para>So that changes in Endpoint Address don't prevent a connection to another node
+        /// in the event they have the Endpoint Address that used to belong to this node</para>
+        /// </summary>
```

Comment:
This does not make much sense to me. We don't really store this information to disk, therefore it will be forgotten after the node restarts. And during the lifetime of the node it would not change it end points.

Reply:
A node running on a dynamic IP address - what happens if the ISP issues a new IP address?

Reply:
That is a good question, but this does not seem to be an answer. As mentioned in the PR comment, I like the fact that we do not need to detect our external endpoints. Therefore suddenly this task is much easier. The disadvantage, however, is that we have no how to react to change of IP address.

We could try to look forward IP address is to other nodes, but then we will lose the benefit of this approach. Therefore I would probably discard the problem of dynamic IP address, because it is unlikely that if we change IP address then on our old IP address another node will exist. So it seems okay not to connect there, pretending that it is our own endpoint.